File: screenshot-tool/query.js
const { Client } = require('pg');
const client = new Client({
  host: 'localhost',
  port: 5432,
  database: 'eThesisProjectDb_dev',
  user: 'postgres',
  password: '1',
});

client.connect()
  .then(async () => {
    console.log('Querying Database for SV0971090 or files...');
    
    // 1. Find User Id for SV0971090
    const userRes = await client.query('SELECT "Id", "FullName" FROM "Users" WHERE "StudentId" = \'SV0971090\'');
    console.log('User:', userRes.rows);

    let queryStr = 'SELECT "Id", "Title", "FilePath", "StudentId" FROM "Theses" WHERE "FilePath" IS NOT NULL';
    if (userRes.rows.length > 0) {
      const userId = userRes.rows[0].Id;
      queryStr += ` OR "StudentId" = ${userId}`;
    }

    const thesesRes = await client.query(queryStr);
    console.log('Theses results:', thesesRes.rows);

    client.end();
  })
  .catch(err => {
    console.error(err);
    client.end();
  });

File: screenshot-tool/query.cs
using System;
using System.Linq;
using Microsoft.EntityFrameworkCore;
using PlatformAdmin.Data;

namespace ScreenshotTool
{
    class Program
    {
        static void Main(string[] args)
        {
            var optionsBuilder = new DbContextOptionsBuilder<AppDbContext>();
            optionsBuilder.UseNpgsql("Host=localhost;Port=5432;Database=eThesisProjectDb_dev;Username=postgres;Password=1");

            using (var context = new AppDbContext(optionsBuilder.Options))
            {
                // 1. Tìm đề tài có chứa file hoặc thuộc về student SV0971090
                var theses = context.Theses
                    .Include(t => t.Student)
                    .Where(t => t.FilePath.Contains("GROUPCHATGPT") || t.Student.StudentId == "SV0971090")
                    .ToList();

                if (theses.Any())
                {
                    Console.WriteLine("FOUND_THESES:");
                    foreach (var t in theses)
                    {
                        Console.WriteLine($"ID: {t.Id} | Title: {t.Title} | FilePath: {t.FilePath} | Student: {t.Student?.FullName} ({t.Student?.StudentId})");
                    }
                }
                else
                {
                    Console.WriteLine("No thesis found for SV0971090 in database.");
                    // In ra một vài đề tài để tham khảo
                    var someTheses = context.Theses.Take(5).ToList();
                    Console.WriteLine("SAMPLE_THESES:");
                    foreach (var t in someTheses)
                    {
                        Console.WriteLine($"ID: {t.Id} | Title: {t.Title} | FilePath: {t.FilePath}");
                    }
                }
            }
        }
    }
}
